esp-hal/src/spi/master.rs

```diff
@@ -2812,6 +2952,22 @@ impl Info {
         Ok(())
     }
 
+    /// Read bytes from SPI.
+    ///
+    /// Sends out a stuffing byte for every byte to read. This function doesn't
+    /// perform flushing. If you want to read the response to something you
+    /// have written before, consider using [`Self::transfer`] instead.
+    #[cfg_attr(place_spi_driver_in_ram, ram)]
+    async fn read_bytes_async(&self, words: &mut [u8]) -> Result<(), Error> {
+        let empty_array = [EMPTY_WRITE_PAD; FIFO_SIZE];
+
+        for chunk in words.chunks_mut(FIFO_SIZE) {
+            self.write_bytes_async(&empty_array[0..chunk.len()]).await?;
+            self.read_bytes_from_fifo(chunk)?;
+        }
+        Ok(())
+    }
```

Comment:
This should be done properly by disabling the write half, rather than wasting CPU cycles writing to the CPU buf.

The blocking API does this already, so it's not important for this PR but it should be done in future.

Reply:
You can see 5 lines up what the blocking API does - which is pretty much the same thing as this function, so I'd appreciate you pointing out what you are thinking of.

Reply:
Yeah, the blocking API is also wasting CPU cycles writing to the CPU buf, as I said in my comment.

https://github.com/esp-rs/esp-hal/blob/bc0bedd628e2149f3111284411d8f5743a9ba8b6/esp-hal/src/spi/master.rs#L2257 is what I'm thinking of.

Reply:
Ah yeah let's leave that for the future.